deployment/kube/prod/frontend.yaml

```diff
@@ -10,22 +10,28 @@ metadata:
     app: imageserver
 spec:
   replicas: 3
+  strategy:                                                                     
+    rollingUpdate:                                                              
+      maxSurge: 100%                                                            
+      maxUnavailable: 1                                                         
+    type: RollingUpdate  
   template:
     metadata:
       labels:
         app: imageserver
         tier: frontend
     spec:
-      affinity:
```

Comment:
Don't we still need antiAffinity to prevent two pods from ending up on the same node? Or are you counting on resource limits to do that? (Smaller commits, and explaining your reasoning in the commit comment would help here...)

Reply:
counting on the resource limit to do that.. and if we do switch for larger instances than we won't care if they spin-up to the same server.

Reply:
When I designed this initially I couldn't find a way to set the resource requests such that one and only one imageserver process could occupy a node, but also allow Kubernetes internal pods to occupy that node.

Is there a way to do it now? This will be an issue if we want to have imageservers in the default node pool, and I think we do.

Reply:
its the case now that's why i'm using a new pool like we have for redis. This way kube-system won't be allow to run there so we won't affect our critical pods.

The only way i saw it done was with toleration + taint

I don't see why we would want imageservers to run on the default pool. Any reason?

Reply:
I explained my concerns about adding a new node pool briefly in the last paragraph here: https://github.com/plotly/orca/pull/128#pullrequestreview-156163750

You don't need tolerations and taints to prevent two pods from occupying the same node. That's what the `podAntiAffinity` statement you're removing was doing, and it was working.

Reply:
- If I use pod `podAntiAffinity` to prevent imageserver on kube-system. Will end-up with the same having dedicated machine for kube-system but sharing the same pool... I don't mind doing. Just simpler to use a pool "backend" and put everything else related in there.

- If I set an `podAntiAffinity` for imageserver we must make sure to apply it on every app this is where I think a pool make sense.

- Another situation is when it scale down it evict the kube-system service that is on the node and it restart else where in case of heapster we lose 5 min of metrics. Not a big deal but we might end up with lots of gap in our graph

Reply:
I'm just suggesting using `podAntiAffinity` to prevent two imageserver pods from occupying the same node like we do now.

I still don't understand why having Kubernetes internal pods on the same nodes as the imageserver pods is a big deal. They don't use significant amounts of resources, do they? I do understand your concern about losing metrics, but I think it's worth trying anyway. I'm surprised Kubernetes isn't designed to scale down by terminating other nodes rather than these, but it sounds like something we have to live with.